Sources/Document/Json/JSONArray.swift

```diff
@@ -16,16 +16,545 @@
+        let removed = try self.target.remove(createdAt: createdAt, executedAt: ticket)
+
+        let operation = RemoveOperation(parentCreatedAt: self.target.getCreatedAt(), createdAt: removed.getCreatedAt(), executedAt: ticket)
+        self.context.push(operation: operation)
+        self.context.registerRemovedElement(removed)
+
+        return removed
+    }
+
+    /**
+     * `splice` is a method to remove elements from the array.
+     */
+    func splice(start: Int, deleteCount: Int? = nil, items: Any...) throws -> [Any] {
+        let length = self.target.length
+        let from = start >= 0 ? Swift.min(start, length) : Swift.max(length + start, 0)
+
+        let to: Int
+        if let deleteCount {
+            if deleteCount < 0 {
+                to = from
+            } else {
+                to = Swift.min(from + deleteCount, length)
+            }
+        } else {
+            to = length
+        }
+
+        var removeds: [Any] = []
+
+        for _ in from ..< to {
+            if let removed = removeInternal(byIndex: from),
+               let element = toJSONElement(from: removed)
+            {
+                removeds.append(element)
+            }
+        }
+
+        if items.isEmpty == false {
+            var previousID: TimeTicket
+            if from == 0 {
+                previousID = self.target.getHead().getID()
+            } else {
+                previousID = try self.target.get(index: from - 1).getID()
+            }
+
+            for item in items {
+                let newElement = try insertAfterInternal(previousCreatedAt: previousID, value: item)
+                previousID = newElement.getID()
+            }
+        }
+
+        return removeds
+    }
+
+    private func adaptedFromIndex(length: Int, fromIndex: Int?) -> Int {
+        let result: Int
+        if let fromIndex {
+            if fromIndex < 0 {
+                result = Swift.max(fromIndex + length, 0)
+            } else {
+                result = fromIndex
+            }
+        } else {
+            result = 0
+        }
+
+        return result
+    }
+
+    /**
+     * `includes` returns true if the given element is in the array.
+     */
+    func includes(searchElement: Any, fromIndex: Int? = nil) -> Bool {
+        let length = self.target.length
+
+        let from = self.adaptedFromIndex(length: length, fromIndex: fromIndex)
+
+        if from >= length {
+            return false
+        }
+
+        if let searchValue = Primitive.type(of: searchElement) {
+            let array = self.target.compactMap { toJSONElement(from: $0) }
+            let searchRange = Array(array[from...])
+            return searchRange.contains { element in
+                guard let value = Primitive.type(of: element) else {
+                    return false
+                }
+
+                return value == searchValue
+            }
+        }
+
+        guard let searchId = (searchElement as? JSONDatable)?.crdtElement.getID() else {
```

Comment:
@hackerwins 
In lastIndexOf, indexOf, and includes functions of ArrayProxy, ```searchElement``` is JSONElement, but it cast ```searchElement``` as CRDTElement. Is that right?
Is the code below correct?
```
        target.getByIndex(i)?.getID() === (searchElement as CRDTElement).getID()
```